Add history items display state for collections/lists with failed or new populated_state

ahmedhamidawan opened the pull request:
Add a non-DB, purely visual state for lists/collections that have `populated_state: failed` ( Fixes #17001 ) or `populated_state: new` ( Fixes #16858 ) in the `HistoryPanel`:

| In the History Panel | In the Collection Panel |
| -------------------- | ----------------------- |
| ![image](https://github.com/galaxyproject/galaxy/assets/78516064/012bd218-c42e-4a89-a707-85fccd7e696f) | ![image](https://github.com/galaxyproject/galaxy/assets/78516064/be6e6087-d93e-47a6-bfa2-abc8a6f22818) |

| In the `StatesInfo` modal found next to the State filter in the History filter menu |
| --------------------------------------------------------------------------------- |
| ![image](https://github.com/galaxyproject/galaxy/assets/78516064/98a91791-65bf-438a-9d30-1aa890eaa62b) |

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

ahmedhamidawan committed:
Add history items display state for empty collections/lists

mvdbeek commented on client/src/components/History/Content/ContentItem.vue:
```diff
@@ -169,6 +173,8 @@ export default {
         state() {
             if (this.isPlaceholder) {
                 return "placeholder";
+            } else if (this.item.element_count !== undefined && this.item.element_count === 0) {
```
That is wrong, sorry. 0 elements is green.

ahmedhamidawan replied:
Ah I see, now I understand, I have also seen the comment on the other issue, now will look for a collection that is not "populated " instead

ahmedhamidawan replied:
Changed it so if we have a failed `populated_state`, we set the item to the newly created `failed_populated_state` state, which has the `danger` variant. Empty collections are still green

mvdbeek commented on client/src/components/History/Content/model/states.ts:
```diff
@@ -103,6 +104,14 @@ export const STATES: StateMap = {
+    /** the collection is empty. This state is only visual and transitional, it does not exist in the database. */
+    emptyCollection: {
+        status: "secondary",
+        text: "This is an empty list/collection.",
```
For what you want to do this is also not correct. You want to highlight a collection that is not populated, that is not the same as being empty.

ahmedhamidawan committed:
change `emptyCollection` item state to `failed_populated_state`

Instead of creating a new, non-ok visual state for empty collections,
create it for collections that have `populated_state: failed`.
Also, show the `populated_state_message` in the `CollectionPanel`

ahmedhamidawan committed:
show `populated_state_message` on hover on icon

ahmedhamidawan committed:
also add a `new_populated_state` for `populated_state: new`

mvdbeek commented on client/src/components/History/Content/model/states.ts:
```diff
@@ -103,6 +104,21 @@ export const STATES: StateMap = {
+    /** the `populated_state: failed`. This state is only visual and transitional, it does not exist in the database. */
+    failed_populated_state: {
+        status: "danger",
+        text: "Failed to populate the list/collection.",
```
```suggestion
        text: "Failed to populate the collection.",
```

ahmedhamidawan committed:
Update client/src/components/History/Content/model/states.ts

Co-authored-by: Marius van den Beek <m.vandenbeek@gmail.com>

mvdbeek commented on client/src/components/History/CurrentCollection/CollectionPanel.vue:
```diff
@@ -115,7 +118,15 @@ watch(
+                        class="m-2"
+                        :variant="populatedStateMsg ? 'danger' : 'info'"
+                        show>
+                        {{ populatedStateMsg || "This is an empty list/collection." }}
```
```suggestion
                        {{ populatedStateMsg || "This is an empty collection." }}
```

ahmedhamidawan committed:
Update client/src/components/History/CurrentCollection/CollectionPanel.vue

Co-authored-by: Marius van den Beek <m.vandenbeek@gmail.com>
